CMakeLists.txt

```diff
@@ -5,9 +5,19 @@ project(cpp-statsd-client
         LANGUAGES CXX
         DESCRIPTION "A header-only StatsD client implemented in C++"
         HOMEPAGE_URL "https://github.com/vthiery/cpp-statsd-client")
+
+option(CPP_STATSD_STANDALONE "Allows configuration of targets for verifying library functionality" ON)
+option(ENABLE_TESTS "Build tests" ON)
+option(ENABLE_COVERAGE "Build with coverage instrumentalisation" OFF)
+
+if(NOT CPP_STATSD_STANDALONE)
+  set(ENABLE_TESTS OFF)
+  set(ENABLE_COVERAGE OFF)
+endif()
```

Comment:
ive added this option to disable the creation of any special  targets for testing the library in standalone mode. this is useful when a project vendors this library in their cmake as they can simply do:

```cmake
SET(CPP_STATSD_STANDALONE OFF CACHE BOOL "ignore test targets" FORCE)
add_subdirectory(${CMAKE_SOURCE_DIR}/thirdparty/cpp-statsd-client ${CMAKE_BINARY_DIR}/thirdparty/cpp-statsd-client)
```

and then they can just depend on the `cpp-statsd-client` target in their targets

Reply:
with out this its highly likely that the targets we create will collide with downstream targets since they are named `check` and `coverage` which are pretty generic